src/app/index/metadata.rs

```diff
@@ -79,6 +94,46 @@ pub fn read(path: &Path) -> Option<SongTags> {
 	}
 }
 
+trait FrameContent {
+	/// Returns the value stored, if any, in the Frame.
+	/// Say "TCOM" returns composer field.
+	fn get_text(&self, key: &str) -> Option<String>;
+
+	/// Returns value stored for `field_name` in custom field `TXXX`.
+	fn get_custom_field(&self, field_name: &str) -> Option<String>;
+}
+
+impl FrameContent for id3::Tag {
+	fn get_text(&self, key: &str) -> Option<String> {
+		let frame = self.get(key)?;
+		match frame.content() {
+			id3::Content::Text(value) => Some(value.to_string()),
+			_ => None,
+		}
+	}
+
+	fn get_custom_field(&self, field_name: &str) -> Option<String> {
```

Comment:
If we do end up keeping the category field, I would prefer writing this block using `match` instead of `unwrap` calls.

Reply:
Dropped the entire function.